### PR TITLE
[Misc] Rename number_of_edges and number_of_nodes to num_edges and num_nodes.

### PR DESCRIPTION
**Pull request by frozenbugs**
## Description

Also renamed the method in [dgl/python/dgl/graph_index.py](https://github.com/dmlc/dgl/blob/4cf5f682a65295ec10c3419cecdcec4388a2235f/python/dgl/graph_index.py#L164) and [dgl/python/dgl/heterograph_index.py](https://github.com/dmlc/dgl/blob/4cf5f682a65295ec10c3419cecdcec4388a2235f/python/dgl/heterograph_index.py#L362).

Such that for all dgl core code, we have consistent name for num_edges and num_nodes.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

**Comment by frozenbugs** [comment]
Should we also rename https://github.com/dmlc/dgl/blob/4cf5f682a65295ec10c3419cecdcec4388a2235f/python/dgl/graph_index.py#L164 and https://github.com/dmlc/dgl/blob/4cf5f682a65295ec10c3419cecdcec4388a2235f/python/dgl/heterograph_index.py#L362?

**Comment by mufeili** [comment]
This PR is too large to review. Note that some examples have their own wrappers of graph data and do not necessarily support `num_nodes`/`num_edges`. Please break it into smaller PRs.

**Comment by mufeili** [comment]
> Should we also rename
> 
> https://github.com/dmlc/dgl/blob/4cf5f682a65295ec10c3419cecdcec4388a2235f/python/dgl/graph_index.py#L164
> 
> and
> https://github.com/dmlc/dgl/blob/4cf5f682a65295ec10c3419cecdcec4388a2235f/python/dgl/heterograph_index.py#L362
> 
> ?

Let's not touch this part for now.

File: python/dgl/graph_index.py [hunk 1]
```diff
@@ -1160,16 +1157,16 @@ def from_networkx(nx_graph, readonly):
             # to_directed creates a deep copy of the networkx graph even if
             # the original graph is already directed and we do not want to do it.
             nx_graph = nx_graph.to_directed()
-    num_nodes = nx_graph.number_of_nodes()
```
Comment (mufeili) [inline python/dgl/graph_index.py:1163]: I don't think NetworkX DiGraph objects have an `num_nodes` API. Could you double check?

File: python/dgl/graph_index.py [hunk 2]
```diff
@@ -1160,16 +1157,16 @@ def from_networkx(nx_graph, readonly):
...
+    num_nodes = nx_graph.num_nodes()
 
     # nx_graph.edges(data=True) returns src, dst, attr_dict
-    if nx_graph.number_of_edges() > 0:
```
Comment (mufeili) [inline python/dgl/graph_index.py:1166]: Ditto

File: python/dgl/graph_index.py [hunk 3]
```diff
@@ -1160,16 +1157,16 @@ def from_networkx(nx_graph, readonly):
...
+    if nx_graph.num_edges() > 0:
         has_edge_id = "id" in next(iter(nx_graph.edges(data=True)))[-1]
     else:
         has_edge_id = False
 
     if has_edge_id:
-        num_edges = nx_graph.number_of_edges()
```
Comment (mufeili) [inline python/dgl/graph_index.py:1172]: Ditto

File: python/dgl/graph_index.py [hunk 4]
```diff
@@ -1182,7 +1179,7 @@ def from_networkx(nx_graph, readonly):
         for e in nx_graph.edges:
             src.append(e[0])
             dst.append(e[1])
-    num_nodes = nx_graph.number_of_nodes()
```
Comment (mufeili) [inline python/dgl/graph_index.py:1185]: Ditto

File: python/dgl/graph_index.py (hunk 1 above)
Comment (frozenbugs) [reply python/dgl/graph_index.py]: Good catch, I reverted many of this mis-change, but this is missed.
Likely due to lack of test coverage.

Comment (mufeili) [reply python/dgl/graph_index.py]: maybe not necessarily in this PR, perhaps we want to examine the functionality of these APIs and see if we should add unit tests to cover them

**Review by mufeili — APPROVED** [review]

**Pull request merged**